runtime-modules/content/src/lib.rs

```diff
@@ -0,0 +1,1142 @@
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum NewAsset<ContentParameters> {
+    /// Upload to the storage system
+    Upload(ContentParameters),
+    /// A url string pointing at an asset
+    Uri(Vec<u8>),
+}
+
+/// The owner of a channel, is the authorized "actor" that can update
+/// or delete or transfer a channel and its contents.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum ChannelOwner<MemberId, CuratorGroupId, DAOId> {
+    /// Do not use - Default value representing empty value
+    Nobody,
+    /// A Member owns the channel
+    Member(MemberId),
+    /// A specific curation group owns the channel
+    CuratorGroup(CuratorGroupId),
+    // Native DAO owns the channel
+    Dao(DAOId),
+}
+
+// Default trait implemented only because its used in a Channel which needs to implement a Default trait
+// since it is a StorageValue.
+impl<MemberId, CuratorGroupId, DAOId> Default for ChannelOwner<MemberId, CuratorGroupId, DAOId> {
+    fn default() -> Self {
+        ChannelOwner::Nobody
+    }
+}
+
+/// A category which channels can belong to. The category will not be deleted if it contains any channels.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ChannelCategory {
+    /// The number of channels in this category.
+    number_of_channels_in: u32,
+}
+
+/// Information on the category being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ChannelCategoryCreationParameters {
+    /// Metadata for the category.
+    meta: Vec<u8>,
+}
+
+/// Information on the category being updated.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ChannelCategoryUpdateParameters {
+    // as this is the only field it is not an Option
+    /// Metadata update for the category.
+    new_meta: Vec<u8>,
+}
+
+/// Type representing an owned channel which videos, playlists, and series can belong to.
+/// If a channel is deleted, all videos, playlists and series will also be deleted.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct ChannelInternal<MemberId, CuratorGroupId, ChannelCategoryId, DAOId, Balance> {
+    /// The owner of a channel
+    owner: ChannelOwner<MemberId, CuratorGroupId, DAOId>,
+    /// The category the channel belongs to
+    in_category: ChannelCategoryId,
+    /// The number of videos under this channel
+    number_of_videos: u32,
+    /// The number of playlists under this channel
+    number_of_playlists: u32,
+    /// The number of series under this channel
+    number_of_series: u32,
+    /// If curators have curated this channel or not
+    is_curated: bool,
```

Comment:
It would be great if we could use something other than `curated`, that really sounds like something hand-picked to be displayed more prominently. `censored` is straightforward and sounds much better to me

Reply:
Okat that also gives completely different meaning to what this field represents. I interpreted curated to mean something that was checked by curators and found to be of good quality and worth promoting over other non curated content. So clearly getting rid of this vague/ambiguous field makes sense.

And I agree censored is obviously quite clear on what it means.

Reply:
done in ebdcd7f326692296d27c4d0baab3731a5e6e9bfc